File: frontend/src/services/appointmentService.js
import api, { unwrapApiResponse } from './api';

const sanitizeAppointmentPayload = (payload = {}) => ({
  patientId: payload.patientId ? Number(payload.patientId) : null,
  doctorId: payload.doctorId ? Number(payload.doctorId) : null,
  appointmentDateTime: payload.appointmentDateTime,
  reason: payload.reason?.trim(),
  notes: payload.notes?.trim() || null,
});

const requestData = async (request) => unwrapApiResponse(await request);

export const getAllAppointments = async () => requestData(api.get('/api/appointments'));
export const getAppointments = getAllAppointments;
export const getAppointmentById = async (id) => requestData(api.get(`/api/appointments/${id}`));
export const bookAppointment = async (payload) => requestData(api.post('/api/appointments', sanitizeAppointmentPayload(payload)));
export const rescheduleAppointment = async (id, payload) =>
  requestData(api.put(`/api/appointments/reschedule/${id}`, sanitizeAppointmentPayload(payload)));
export const cancelAppointment = async (id) => requestData(api.put(`/api/appointments/cancel/${id}`));

File: backend/src/main/java/com/hms/service/AppointmentService.java
package com.hms.service;

import java.util.List;

import com.hms.dto.AppointmentRequestDTO;
import com.hms.dto.AppointmentResponseDTO;

public interface AppointmentService {

    AppointmentResponseDTO bookAppointment(AppointmentRequestDTO request);

    AppointmentResponseDTO rescheduleAppointment(Long id, AppointmentRequestDTO request);

    AppointmentResponseDTO cancelAppointment(Long id);

    List<AppointmentResponseDTO> getAllAppointments();

    AppointmentResponseDTO getAppointmentById(Long id);
}
